packages/@aws-cdk/app-delivery/test/test.pipeline-deploy-stack-action.ts

```diff
@@ -51,14 +62,170 @@ export = nodeunit.testCase({
+    })));
+    expect(pipelineStack).to(haveResource('AWS::CodePipeline::Pipeline', hasPipelineAction({
+      Configuration: {
+        StackName: "NoCapStack",
+        ActionMode: "CHANGE_SET_REPLACE",
+      }
+    })));
+    test.done();
+  },
+  'users can supply enable full permissions'(test: nodeunit.Test) {
+    const pipelineStack = getTestStack();
+    const selfUpdatingStack = createSelfUpdatingStack(pipelineStack);
+
+    const pipeline = selfUpdatingStack.pipeline;
+    const selfUpdateStage = pipeline.addStage('SelfUpdate');
+    new PipelineDeployStackAction(pipelineStack, 'SelfUpdatePipeline', {
+      stage: selfUpdateStage,
+      stack: pipelineStack,
+      inputArtifact: selfUpdatingStack.synthesizedApp,
+      adminPermissions: true,
+    });
+    expect(pipelineStack).to(haveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: '*',
+            Effect: 'Allow',
+            Resource: '*',
+          }
+        ],
+      }
+    }));
+    test.done();
+  },
+  'users can supply a role for deploy action'(test: nodeunit.Test) {
+    const pipelineStack = getTestStack();
+    const selfUpdatingStack = createSelfUpdatingStack(pipelineStack);
+
+    const pipeline = selfUpdatingStack.pipeline;
+    const role = new iam.Role(pipelineStack, 'MyRole', {
+      assumedBy: new iam.ServicePrincipal('cloudformation.amazonaws.com'),
+    });
+    const selfUpdateStage = pipeline.addStage('SelfUpdate');
+    const deployAction = new PipelineDeployStackAction(pipelineStack, 'SelfUpdatePipeline', {
+      stage: selfUpdateStage,
+      stack: pipelineStack,
+      inputArtifact: selfUpdatingStack.synthesizedApp,
+      adminPermissions: false,
+      role
+    });
+    test.deepEqual(role.id, deployAction.role.id);
+    test.done();
+  },
+  'users can specify IAM permissions for the deploy action'(test: nodeunit.Test) {
+    // GIVEN //
+    const pipelineStack = getTestStack();
+
+    // the fake stack to deploy
+    const emptyStack = getTestStack();
+
+    const selfUpdatingStack = createSelfUpdatingStack(pipelineStack);
+    const pipeline = selfUpdatingStack.pipeline;
+
+    // WHEN //
+    // this our app/service/infra to deploy
+    const deployStage = pipeline.addStage('Deploy');
+    const deployAction = new PipelineDeployStackAction(pipelineStack, 'DeployServiceStackA', {
+      stage: deployStage,
+      stack: emptyStack,
+      inputArtifact: selfUpdatingStack.synthesizedApp,
+      adminPermissions: false,
+    });
+    // we might need to add permissions
+    deployAction.addToRolePolicy( new iam.PolicyStatement().
+      addAction('ec2:AuthorizeSecurityGroupEgress').
+      addAction('ec2:AuthorizeSecurityGroupIngress').
+      addAction('ec2:DeleteSecurityGroup').
+      addAction('ec2:DescribeSecurityGroups').
+      addAction('ec2:CreateSecurityGroup').
+      addAction('ec2:RevokeSecurityGroupEgress').
+      addAction('ec2:RevokeSecurityGroupIngress').
```

Comment:
Feel free to use the `addActions` method here, that takes a varargs argument, will be more concise.

Reply:
OH... I didn't see there was one!